apps/dashboard/app/controllers/projects_controller.rb

```diff
@@ -1,45 +1,63 @@
 # frozen_string_literal: true
 
 class ProjectsController < ApplicationController
+  
+  # GET /projects/:id
   def show
     # files in current project
   end
 
+  # GET /projects
   def index
     @projects = Project.all
   end
 
+  # GET /projects/new
   def new
     @project = Project.new
   end
 
+  # GET /projects/:id/edit
   def edit
     @project = Project.find(params[:id])
   end
 
+  # PATCH /projects/:id
+  def update
+    @project = Project.find(params[:id])
+
+    if @project.update(project_params)
+      redirect_to projects_path, notice: 'Project manifest updated!'
+    end
+  end
+
+  # POST /projects
   def create
     @project = Project.new(project_params)
 
-    if @project.save!
-      @project.config_dir
+    # saving .ondemand in project_dataroot 
+    if @project.valid? 
+      @project.save
```

Comment:
This'll capture errors on save.

```suggestion
    if @project.valid? && @project.save
```